client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/MethodInvocation.java

```diff
@@ -116,11 +119,13 @@ public boolean hasNestedParameters(String path) {
     }
 
     private Stream<ParameterInfo> parameters() {
-        Parameter[] parameters = method.getParameters();
-        return IntStream.range(0, parameters.length)
-                .mapToObj(i -> new ParameterInfo(this,
-                        parameters[i],
-                        parameterValues[i]));
+        if (this.parameters == null)
+            this.parameters = IntStream.range(0, method.getParameterCount())
+                    .mapToObj(i -> new ParameterInfo(this,
+                            method.getParameters()[i],
+                            parameterValues[i]))
+                    .collect(toList());
+        return parameters.stream();
```

Comment:
The query will be cached, but the parameters are accessed many times, so this should improve warmup times